sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp

```diff
@@ -62,23 +62,23 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
      * @param directoryUri The URI of the file system this client's request targets.
      * @param options Optional parameters used to initialize the client.
      */
-    explicit DirectoryClient(
+    explicit DataLakeDirectoryClient(
         const std::string& directoryUri,
         const DataLakeClientOptions& options = DataLakeClientOptions());
 
     /**
-     * @brief Create a FileClient from current DirectoryClient
+     * @brief Create a FileClient from current DataLakeDirectoryClient
      * @param path Path of the file under the directory.
      * @return FileClient
      */
-    FileClient GetFileClient(const std::string& path) const;
+    DataLakeFileClient GetDataLakeFileClient(const std::string& path) const;
 
     /**
-     * @brief Create a DirectoryClient from current DirectoryClient
+     * @brief Create a DataLakeDirectoryClient from current DataLakeDirectoryClient
      * @param path Path of the directory under the current directory.
-     * @return DirectoryClient
+     * @return DataLakeDirectoryClient
      */
-    DirectoryClient GetSubDirectoryClient(const std::string& path) const;
+    DataLakeDirectoryClient GetSubDataLakeDirectoryClient(const std::string& path) const;
```

Comment:
```suggestion
    DataLakeDirectoryClient GetSubDirectoryClient(const std::string& path) const;
```

Reply:
It's `GetSubShareDirectoryClient` in file share.

Reply:
So does `Sub` modify `DataLake` or `Directory`? I know what subdirectory is, but what is a **subdatalake**?

Reply:
Did you just globally search and replace these words?

Reply:
No, it is not `SubDataLake Directory`, it is `Sub DataLakeDirectory`. Anyway as discussed, I'll make these `GetSubDirectoryClient`.

Reply:
I still think it should be `DataLakeSubdirectory` as `subdirectory` is one word. You cannot split one word